src/EFCore.PG/Storage/Internal/NpgsqlTypeMappingSource.cs

```diff
@@ -816,6 +760,57 @@ static Type FindTypeToInstantiate(Type collectionType, Type elementType)
                 ? new NpgsqlRowValueTypeMapping(clrType)
                 : null;
 
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    protected virtual RelationalTypeMapping? FindEnumMapping(in RelationalTypeMappingInfo mappingInfo)
+    {
+        var storeType = mappingInfo.StoreTypeName;
+        var clrType = mappingInfo.ClrType;
+
+        if (clrType is not null and not { IsEnum: true, IsClass: false })
+        {
+            return null;
+        }
+
+        // Try to find an enum definition (defined by the user on their context options), based on the
+        // incoming MappingInfo's StoreType or ClrType
+        EnumDefinition? enumDefinition;
+        if (storeType is null)
+        {
+            enumDefinition = _enumDefinitions.SingleOrDefault(m => m.ClrType == clrType);
+        }
+        else
+        {
+            // TODO: Not sure what to do about quoting. Is the user expected to configure properties
```

Comment:
TODO

Reply:
Do you think the current code as-is is good enough for merging for the moment (e.g. until we do https://github.com/npgsql/npgsql/issues/5710)?

Reply:
Yeah as long as users can pass quoted names - as I understand they can - they should always be able to make EF map things correctly when mappingInfo.StoreTypeName is quoted.